### PR TITLE
topology2: add basic definitions for PTL

### PR DESCRIPTION
**Pull request by plbossart**
Start moving SoundWire topologies from the shadows into main...

### CODE REVIEW

**Comment by plbossart** [comment]
@bardliao any idea how to create sof-ptl-rt722-l3 ?

**Comment by fredoh9** [comment]
@bardliao built and generated it, I was able to test it on the FPGA.

File: tools/topology/topology2/platform/intel/ptl.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# PTL-specific variable definitions
+Define {
+	DMIC_DRIVER_VERSION		5
+	SSP_BLOB_VERSION		0x300
+	NUM_HDMIS			3
+}
```
Comment (marc-hb) [inline tools/topology/topology2/platform/intel/ptl.conf:6]: This is duplicated in #9306, just a heads up.
cc: @singalsu

File: tools/topology/topology2/cavs-sdw.conf [hunk 2]
```diff
@@ -85,6 +85,7 @@ Define {
 IncludeByKey.PLATFORM {
 	"mtl"	"platform/intel/mtl.conf"
 	"lnl"	"platform/intel/lnl.conf"
+	"ptl"	"platform/intel/ptl.conf"
```
Comment (marc-hb) [inline tools/topology/topology2/cavs-sdw.conf:88]: Probably a dumb question sorry: why is this file called CAVS-sdw.conf?

Comment (plbossart) [reply tools/topology/topology2/cavs-sdw.conf:88]: because we started using it on cavs... it's meaningless, only the sdw part matters.

**Comment by bardliao** [comment]
> @bardliao @fredoh9 any idea how to create sof-ptl-rt722-l3 ?

I believe that rt722-l3 can use the same topology as rt722-l0. We use partial match to match the stream name and do not include the link information. I.e. `SDW_SPK_STREAM=Playback-SmartAmp,SDW_DMIC_STREAM=Capture-SmartMic,\
SDW_JACK_OUT_STREAM=Playback-SimpleJack,SDW_JACK_IN_STREAM=Capture-SimpleJack`

**Comment by plbossart** [comment]
then why do we have different topology names in the kernel? We could just as well use sof-ptl-rt722.tplg in all ACPI tables where RT722 is used.

**Comment by plbossart** [comment]
v2: used a single topology name. The matching kernel change is here: https://github.com/thesofproject/linux/pull/5111

**Comment by marc-hb** [comment]
SOFCI TEST

**Review by ujfalusi — APPROVED** [review]

File: tools/topology/topology2/production/tplg-targets-ace3.cmake [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+# Array of "input-file-name;output-file-name;comma separated pre-processor variables"
+list(APPEND TPLGS
+# SDW topology for PTL RVP
+"cavs-sdw\;sof-ptl-rt722\;PLATFORM=ptl,SDW_DMIC=1,NUM_SDW_AMP_LINKS=1,\
```
Comment (ujfalusi) [inline tools/topology/topology2/production/tplg-targets-ace3.cmake:6]: I guess this has dependency on https://github.com/thesofproject/linux/pull/5111 ?

Comment (plbossart) [reply tools/topology/topology2/production/tplg-targets-ace3.cmake:6]: yep, it's even described above :-)

**Comment by marc-hb** [comment]
SOFCI TEST

File: tools/topology/topology2/platform/intel/ptl.conf (hunk 1 above)
Comment (singalsu) [reply tools/topology/topology2/platform/intel/ptl.conf:6]: Yep, same definitions to enable build of PR. Rebase will fix this whichever PR is merged first.

**Review by singalsu — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**